File: app/src/main/java/com/comicsopentrends/fragments/mvp/clans/presenter/OnFinishClansListener.java
package com.comicsopentrends.fragments.mvp.clans.presenter;

import com.comicsopentrends.model.ResponseClans;

public interface OnFinishClansListener {
    void onFailureSearchClan(Throwable throwable);
    void onSuccessSearchClan(ResponseClans response);

    void successClans(ResponseClans response);
    void onCompleteClans();
    void onFailureClans(Throwable throwable);
}

File: app/src/main/java/com/comicsopentrends/fragments/mvp/clans/presenter/OnFinishClansListener.kt
package com.comicsopentrends.fragments.mvp.clans.presenter

import com.comicsopentrends.model.ResponseClans

interface OnFinishClansListener {
    fun onFailureSearchClan(throwable: Throwable)
    fun onSuccessSearchClan(response: ResponseClans)

    fun successClans(response: ResponseClans)
    fun onCompleteClans()
    fun onFailureClans(throwable: Throwable)
}
